src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -219,6 +219,12 @@ public function buildView(FormView $view, FormInterface $form, array $options)
             // POST request.
             $view->vars['full_name'] .= '[]';
         }
+
+        // Provide a default message when choices var is empty
+        if (empty($view->vars['choices']) && $options['expanded']) {
+            $view->vars['empty_choices'] = null !== $options['empty_choices']
+                ? $options['empty_choices'] : '<em>No choice available</em>';
```

Comment:
Please do not mix messages and view data (which actually required you to use the `raw` filter in the form themes). Instead, you should only configure a message here and pass it to the `trans` filter in the form themes (by the way, you currently miss translating the message).